Makefile

```diff
@@ -503,14 +503,14 @@ gen-doc =	\
 		else \
 			cd tools/doc && node ../../$(NPM) install; \
 		fi;\
-	[ -x $(NODE) ] && $(NODE) $(gen-json) || node
+	[ -x $(NODE) ] && $(NODE) $(1) || node
```

Comment:
Bearing in mind I'm a Makefile novice, when I have this diff:

```diff
diff --git a/Makefile b/Makefile
index 91466e1003..384164e24d 100644
--- a/Makefile
+++ b/Makefile
@@ -503,14 +503,14 @@ gen-doc = \
                else \
                        cd tools/doc && node ../../$(NPM) install; \
                fi;\
-       [ -x $(NODE) ] && $(NODE) $(1) || node
+       [ -x $(NODE) ] && $(NODE) || node
 
 out/doc/api/%.json: doc/api/%.md
-       $(call gen-doc, $(gen-json))
+       $(gen-doc) $(gen-json)
 
 # check if ./node is actually set, else use user pre-installed binary
 out/doc/api/%.html: doc/api/%.md
-       $(call gen-doc, $(gen-html))
+       $(gen-doc) $(gen-html)
 
 docopen: $(apidocs_html)
        @$(PYTHON) -mwebbrowser file://$(PWD)/out/doc/api/all.html
```

It causes this to be executed (this is a long makefile command, I omitted non-relevant sections):

```bash
[ -x ./node ] && ./node || node tools/doc/generate.js --node-version=v9.0.0 --format=html --template=doc/template.html --analytics= doc/api/_toc.md > out/doc/api/_toc.html
```

It looks like the $(gen-json/html) arg isn't being passed to `$(NODE)`, but is being passed to `node`. Would appreciate any suggestions for how to fix this (I've already tried some SO and Google searches) 😞